examples/benchmarks/train_resnet_imagenet1k.py

```diff
@@ -0,0 +1,320 @@
+)
+eval_dataspec = DataSpec(dataloader=eval_dataloader,
+                         device_transforms=NormalizationFn(mean=IMAGENET_CHANNEL_MEAN, std=IMAGENET_CHANNEL_STD))
+logging.info('Built evaluation dataloader')
+
+# Instantiate torchvision ResNet model
+logging.info('Build Composer model')
+model_fn = getattr(resnet, args.model_name)
+model = model_fn(num_classes=1000, groups=1, width_per_group=64)
+
+
+# Specify model initialization
+def weight_init(w: torch.nn.Module):
+    if isinstance(w, torch.nn.Linear) or isinstance(w, torch.nn.Conv2d):
+        torch.nn.init.kaiming_normal_(w.weight)
+    if isinstance(w, torch.nn.BatchNorm2d):
+        w.weight.data = torch.rand(w.weight.data.shape)
+        w.bias.data = torch.zeros_like(w.bias.data)
+    # When using binary cross entropy, set the classification layer bias to -log(num_classes)
+    # to ensure the initial probabilities are approximately 1 / num_classes
+    if args.loss_name == 'binary_cross_entropy' and isinstance(w, torch.nn.Linear):
+        w.bias.data = torch.ones(w.bias.shape) * -torch.log(torch.tensor(w.bias.shape[0]))
+
+
+model.apply(weight_init)
+
+# Performance metrics to log outside of training loss
+train_metrics = Accuracy()
+val_metrics = MetricCollection([CrossEntropy(), Accuracy()])
+
+# Cross entropy loss that can handle both index and one-hot targets
+
+if args.loss_name == 'binary_cross_entropy':
+    loss_fn = binary_cross_entropy_with_logits
+else:
+    loss_fn = soft_cross_entropy
+
+# Wrapper function to convert a classification PyTorch model into a Composer model
+composer_model = ComposerClassifier(model, train_metrics=train_metrics, val_metrics=val_metrics, loss_fn=loss_fn)
+logging.info('Built Composer model')
+
+# Optimizer
+logging.info('Build optimizer and learning rate scheduler')
+optimizer = DecoupledSGDW(composer_model.parameters(),
+                          lr=args.learning_rate,
+                          momentum=args.momentum,
+                          weight_decay=args.weight_decay)
+
+# Learning rate scheduler: LR warmup for 8 epochs, then cosine decay for the rest of training
+lr_scheduler = CosineAnnealingWithWarmupScheduler(t_warmup=args.t_warmup, t_max=args.t_max, alpha_f=args.alpha_f)
+logging.info('Build optimizer and learning rate scheduler')
+
+# Callbacks for logging
+logging.info('Build SpeedMonitor, LRMonitor, and CheckpointSaver callbacks')
+speed_monitor = SpeedMonitor(window_size=50)
+lr_monitor = LRMonitor()
+
+# Callback for checkpointing
+checkpoint_saver = CheckpointSaver(folder=args.save_checkpoint_dir, save_interval=args.checkpoint_interval)
+logging.info('Build SpeedMonitor, LRMonitor, and CheckpointSaver callbacks')
+
+# Recipes for training ResNet architectures on ImageNet in order of increasing training time and accuracy
+logging.info('Build algorithm recipes')
+if args.recipe_name == 'mild':
+    algorithms = [
+        BlurPool(),  # Add anti-aliasing filters to strided convs and maxpools
+        ChannelsLast(),
```

Comment:
We should probably either explain all of the algorithms or none of them. And I'm not sure there's a concise way to explain them all. Maybe method card links? If we go that route, might look cleaner to deduplicate these lines, so that there's only one line instantiating a BlurPool(), one for a ChannelsLast(), etc.
```suggestion
        BlurPool(),
        ChannelsLast(),
```

Reply:
Yeah, I was ambitiously going to describe them all, but gave up quickly 😅 What if I add a link to our method card documentation in a comment before the if-statement?